src/main/java/core/basesyntax/service/impl/FruitListImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitTransaction;
+import core.basesyntax.service.FruitList;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FruitListImpl implements FruitList {
+    @Override
+    public List<FruitTransaction> getFruitList(List<FruitTransaction> list) {
+        List<String> listOfFruitsString = list.stream()
+                .map(FruitTransaction::getFruit)
+                .distinct()
+                .toList();
+        List<FruitTransaction> listOfFruitsAsObjects = new ArrayList<>();
+        for (String fruit : listOfFruitsString) {
+            int defaultQuantity = 0;
+            listOfFruitsAsObjects.add(new FruitTransaction(fruit, defaultQuantity));
```

Comment:
The `FruitTransaction` constructor is used with a default quantity of 0. If the default quantity is always 0, consider adding a constructor in `FruitTransaction` that only takes the fruit name and sets the quantity to 0 internally.